### PR TITLE
Support delay in specification

### PR DESCRIPTION
**Pull request by SooluThomas**
### Summary

Added Delay to the qobj schema to support "delay" instructions

### CODE REVIEW

File: qiskit/schemas/qobj_schema.json [hunk 1]
```diff
@@ -153,6 +154,30 @@
...
                     "title": "Frame change pulse"
                 },
+                {
+                    "properties": {
+                        "name": {
+                            "enum": [
+                                "delay"
+                            ]
+                        },
+                        "val": {
```
Comment (taalexander) [inline qiskit/schemas/qobj_schema.json:164]: What is "val"? I also do not see `duration` or `t0` keys defined?

Comment (SooluThomas) [reply qiskit/schemas/qobj_schema.json]: fixed in [44303cd](https://github.com/Qiskit/qiskit-terra/pull/4212/commits/44303cd75e146cf0e793b61c636d0f77f93aed17)

File: qiskit/schemas/qobj_schema.json [hunk 2]
```diff
@@ -153,6 +154,30 @@
...
                     "title": "Frame change pulse"
                 },
+                {
+                    "properties": {
+                        "name": {
+                            "enum": [
+                                "delay"
+                            ]
+                        },
+                        "duration": {
```
Comment (taalexander) [inline qiskit/schemas/qobj_schema.json:164]: Why is the definition of duration so complex? Could it be something similar to how t0 is defined below? 
```json
"t0": {
           "minimum": 0,
            "type": "integer"
                }
```

Comment (SooluThomas) [reply qiskit/schemas/qobj_schema.json:164]: ok. I've updated this to
```
"duration": { type: "integer" }
```

File: qiskit/schemas/qobj_schema.json [hunk 3]
```diff
@@ -153,6 +154,48 @@
...
```
Comment (taalexander) [inline qiskit/schemas/qobj_schema.json:164]: If I look at `acquire` below the definition of duration and other parameters is much simpler. Could we use this form?

**Pull request merged**